src/main/java/com/uid2/admin/vertx/service/ServiceLinkService.java

```diff
@@ -172,7 +199,26 @@ private void handleServiceLinkUpdate(RoutingContext rc) {
                 return;
             }
 
-            serviceLink.setName(name);
+            if (name != null && !name.isEmpty()) {
+                serviceLink.setName(name);
+            }
+
+            if (rolesJson != null && !rolesJson.isEmpty()) {
+                final Set<Role> roles;
+                try {
```

Comment:
This blocks looks very similar to the one in `handleServiceLinkAdd`: do you think it is possible/feasible to reuse the code?